docs/tutorials/hamiltonians.md

```diff
@@ -130,3 +115,34 @@ nn_ham = hamiltonian_factory(
 
 print(nn_ham) # markdown-exec: hide
 ```
+
+Alternatively, fully customizable sympy functions can be passed in an array using the Qadence parameters.
+Furthermore, the `use_complete_graph` option can be passed so that interactions are created for every single
+node pair in the register, irrespectively of the topology of the edges. This is useful for creating Hamiltonians
+that depend on qubit distance.
+
+```python exec="on" source="material-block" result="json" session="hamiltonians"
+from qadence import VariationalParameter, Register
+
+# Square register of 4 qubits with a dimensionless distance of 8.0
+reg = Register.square(2, spacing = 8.0)
+
+# Get the distances between all pairs of qubits
+distance_dict = reg.distances
+
+# Create interaction strength with variational parameter and 1/r term
+strength_list = []
+for edge in reg.all_edges:
```

Comment:
Here I'm just showcasing how you would do it to create an Hamiltonian that depends on the distances between qubits irrespectively of the edges. If the user wants to take just the edges from the register topology into account they could still run this with `for edge in reg.edges` and ommit the `use_complete_graph = True` below. I can add a few more comments to clarify this, but overall it's just about giving the user the option to build whatever Hamiltonian they want.

Reply:
Sure, I get that. What I mean is that the distinction between  the `.edges` and `.all_edges` properties is blurry to me. Maybe this is an issue with the name.

Reply:
Yes I see, I can rethink a bit the naming here. Would you prefer maybe `.all_node_pairs`? Answering also to the comment above, I agree that an edge is a _connected_ node pair, but a node pair in a graph is more general and can also be unconnected.

Reply:
A bit unrelated but I just realized I can remove the auxiliary complete graph from the `Register`, since it is no longer used by `hamiltonian_factory` function. It was previously because this function would save the interaction strengths into the edge properties of the networkx graph, but this changed the register in place and caused issues in other places of the code.